api/migrations/20200709183253-loadMemDocs.js

```diff
@@ -101,44 +104,59 @@ exports.up = async function (db) {
+            if (!existingDoc) {
+              if (collection.type && collection.type.length > 0 && !collection.isForEAO) {
+                try {
+                  const newDoc = await createMineDocument(nrpti, nrptiMine, collection, collectionDoc, bcmiCollection._id);
+                  if (newDoc) {
+                    allNewDocs.push(new ObjectID(newDoc._id));
+                    docsCreated += 1;
+                  } else {
+                    throw Error('No document generated?');
+                  }
+                } catch(err) {
+                  console.error('#######################################');
+                  console.error(`## An error occured while creating the doc`);
+                  console.error(err);
+                  console.error('#######################################');
+                  errors += 1;
                 }
-              } catch(err) {
-                console.error('#######################################');
-                console.error(`## An error occured while creating the doc`);
-                console.error(err);
-                console.error('#######################################');
-                errors += 1;
               }
+            } else {
+              docsExisting += 1;
+            }
+          }
+          const existingCollection  = await nrpti.findOne({ _schemaName: "CollectionBCMI", name: collection.displayName });
+          if (!existingCollection) {
+            console.log(`Creating NRPTI collection for ${collection.displayName}`);
+            // Master, Meta and Documents for this collection are all created.
+            // Now, create a NRPTI collection and shove the docs into it!
+            bcmiCollection._master = new ObjectID(nrptiMine._id);
+            bcmiCollection.read =  [utils.ApplicationRoles.ADMIN, utils.ApplicationRoles.ADMIN_BCMI, 'public'];
+            bcmiCollection.write = [utils.ApplicationRoles.ADMIN, utils.ApplicationRoles.ADMIN_BCMI];
+            bcmiCollection.name = collection.displayName;
+            bcmiCollection.date = collection.date;
+            bcmiCollection.project = new ObjectID(nrptiMine._id);
+            bcmiCollection.type = collection.type;
+            bcmiCollection.agency = collection.isForMEM ? 'EMPR' : collection.isForEAO ? 'EAO' : 'ENV';
+            bcmiCollection.records = allNewDocs; // move this whole thing to the top if we remove records array
+            bcmiCollection.addedBy = 'nrpti';
+            bcmiCollection.datePublished = collection.date;
+            bcmiCollection.publishedBy = 'nrpti';
+            bcmiCollection.isBcmiPublished = true;
+            // todo add lookup for existing collection skip creating if it exists
+            await nrpti.insertOne(bcmiCollection);
+
+            collectionsCreated += 1;
+          } else {
+            if (allNewDocs) {
```

Comment:
The length property will need to be checked. Currently this will always pass.

Reply:
👍